src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
+const ANIMAL_DAMAGE = 50;
+const ANIMAL_BASE_HP = 100;
+const MINIMUM_HP = 0;
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = ANIMAL_BASE_HP;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.constructor === Carnivore || animal.hidden) {
+      return;
+    }
+
+    animal.health -= ANIMAL_DAMAGE;
+
+    if (animal.health <= MINIMUM_HP) {
+      Animal.alive = Animal.alive.filter((a) => a !== animal);
```

Comment:
Perhaps it would be better to move this logic to a separate method `die` (for the Animal class) for better reusability and readability.
  
```
  die() {
    Animal.alive = Animal.alive.filter((a) => a !== this);
  }
```